manifests/common.pp

```diff
@@ -0,0 +1,71 @@
+# @summary Sets up common Kubernetes components - users/groups/folders/etc
```

Comment:
can you please add `# @api private` and call `assert_private()` ?

Reply:
There were a bunch of places where this was incorrect in the module, so I'm pushing all of them at once, expanding the scope of this PR slightly.